### PR TITLE
596 display gas usage

### PR DESCRIPTION
**Pull request by tkumor3**
Closes #

## Introduced changes

-

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Forge
+
+#### Added
+
+- `--print-gas-usage`, short `-g` flag to print gas usage for each test
+
```
Comment (MaksymilianDemitraszek) [inline CHANGELOG.md:15]: Why this is hidden under a flag?

**Review by MaksymilianDemitraszek — CHANGES_REQUESTED** [review]
How does it work for fuzzer? We probably should calculate an average & std deviation.

File: CHANGELOG.md (hunk 1 above)
Comment (cptartur) [reply CHANGELOG.md:15]: I don't think it's useful to always display it

**Review by cptartur — COMMENTED** [review]
Let's update CHANGELOG.md file as well

File: CHANGELOG.md (hunk 1 above)
Comment (piotmag769) [reply CHANGELOG.md:15]: Tbh this flag is weird, I would just always print it

**Review by piotmag769 — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (tkumor3) [reply CHANGELOG.md:15]: If we do that default, we will need to update our all e2e tests 😨

Comment (cptartur) [reply CHANGELOG.md:15]: I don't have a very strong opinion on that. We can always print it if you all think that's useful.

File: crates/forge-runner/src/lib.rs [hunk 2]
```diff
@@ -377,9 +378,36 @@ fn run_with_fuzzing(
...
+                .iter()
+                .filter(|item| matches!(item, TestCaseSummary::Passed { .. }))
+                .map(|a| match a {
+                    TestCaseSummary::Passed { gas_used, .. } => gas_used,
+                    _ => unreachable!(),
+                })
+                .collect();
+
+            let max = gas_usages_vec
+                .clone()
```
[the remark this answers is not in the document]
Comment (tkumor3) [reply crates/forge-runner/src/lib.rs:392]: Zrzut ekranu 2023-12-6 o 08 32 38

Any idea how to handle it?

File: crates/forge-runner/src/lib.rs [hunk 3]
```diff
@@ -377,9 +378,36 @@ fn run_with_fuzzing(
...
+                .into_iter()
+                .copied()
+                .reduce(f64::max)
+                .unwrap();
+            let min = gas_usages_vec
+                .clone()
```
[the remark this answers is not in the document]
Comment (tkumor3) [reply crates/forge-runner/src/lib.rs:398]: this one I can remove, iter_into can take over the ownership

File: crates/forge-runner/src/lib.rs [hunk 4]
```diff
@@ -377,9 +378,36 @@ fn run_with_fuzzing(
...
+                .into_iter()
+                .copied()
+                .reduce(f64::min)
+                .unwrap();
+            //TODO: compute avg
+            let avg: f64 = 3.23;
```
[the remark this answers is not in the document]
Comment (tkumor3) [reply crates/forge-runner/src/lib.rs:404]: I hope that I will be able to finish it before release ;)

**Pull request merged**